xarray/tests/test_backends.py

```diff
@@ -2649,6 +2649,14 @@ def test_chunks(self):
                 ex = expected.sel(band=1).mean(dim='x')
                 assert_allclose(ac, ex)
 
+    def test_pickle_rio(self):
+        # regression test for https://github.com/pydata/xarray/issues/2121
+        with create_tmp_geotiff() as (tmp_file, expected):
+            with xr.open_rasterio(tmp_file) as rioda:
+                temp = pickle.dumps(rioda)
+                actual = pickle.loads(temp)
```

Comment:
Something to think about here -- I'm pretty sure this leaks a file descriptor. I'm not quite sure what the best way to deal with this is (maybe it's not worth worrying about yet), but in the past this has caused issues with tests on various platforms (especially Windows).

Reply:
And indeed, it's causing a test failure on Windows :).

I *think* it should work to explicitly open `actual` in a context manager, e.g.,
```
with pickle.loads(temp) as actual:
    ...
```